src/array/operators.jl

```diff
@@ -111,3 +111,24 @@ end
 
 Base.first(A::DArray) = A[begin]
 Base.last(A::DArray) = A[end]
+
+# In-place operations
+
+function Base.map!(f, a::DArray{T}) where T
+    Dagger.spawn_datadeps() do
+        for ca in vec(chunks(a))
```

Comment:
Out of curiosity, is it necessary to flatten `chunks(a)` like this?

Reply:
Good point! It is not necessary.
```suggestion
        for ca in chunks(a)
```